miden-lib/asm/miden/kernels/tx/account.masm

```diff
@@ -413,62 +429,86 @@ export.authenticate_procedure.1
     # => [PROC_ROOT]
 end
 
-#! Validates that the account seed, provided via the advice map, satisfies the seed requirements.
+#! Validates the account id and its proof-of-work.
 #!
 #! Validation is performed via the following steps:
-#! 1. compute the hash of (SEED, CODE_ROOT, STORAGE_ROOT, 0, 0, 0, 0)
-#! 2. Assert the least significant element of the digest is equal to the account id of the account
-#!    the transaction is being executed against.
-#! 3. Assert the most significant element has sufficient proof of work (trailing zeros) for the account
-#!    type the transaction is being executed against.
 #!
-#! Stack: []
+#! 1. Compute `hash(SEED || CODE_ROOT || [acc_id,0,0,0] || STORAGE_ROOT)`.
+#! 2. Assert the least significant element of the digest is same as the account id.
+#! 3. Assert the most significant element has sufficient proof of work for the account type.
+#!
+#! This approach has the following properties:
+#!
+#! - The account id is generated from a commitment to initial account's state. This prevents changes to
+#!   the account's initial state by a malicious operator.
+#! - The account id is determined by a hash, as long as there is an honest operator including
+#!   transactions in the rollup, an attacker would not have enough time to grind and front run an
+#!   account creation transaction.
+#!   - The cost for an attacker to grind the account id is significantly larger than the cost to create
+#!     an id, because the original id is random and requires little grinding by the honest user, but an
+#!     attacker would need to grind to match the complete account id.
+#! - Account creation is not free, this introduces a small cost to account creation which as a small
+#!   spam proteciton.
+#!
+#! Stack: [CODE_ROOT, SEED, STORAGE_ROOT, acc_id]
```

Comment:
Changed the procedure to receive the data over the stack, so that it can be tested. This only has the cost of two additional instrucitons `swapw` and a `movdnw.2`